lib/jxl/enc_frame.cc

```diff
@@ -1317,13 +1340,42 @@ Status EncodeGroups(const FrameHeader& frame_header,
 
 Status ComputeEncodingData(
     const CompressParams& cparams, const FrameInfo& frame_info,
-    const CodecMetadata* metadata, FrameHeader& frame_header,
-    JxlEncoderChunkedFrameAdapter& frame_data, size_t x0, size_t y0,
-    size_t xsize, size_t ysize, const JxlCmsInterface& cms, ThreadPool* pool,
+    const CodecMetadata* metadata, JxlEncoderChunkedFrameAdapter& frame_data,
+    const jpeg::JPEGData* jpeg_data, size_t x0, size_t y0, size_t xsize,
+    size_t ysize, const JxlCmsInterface& cms, ThreadPool* pool,
+    FrameHeader& mutable_frame_header, ModularFrameEncoder& enc_modular,
     PassesEncoderState& enc_state, std::vector<BitWriter>* group_codes,
     AuxOut* aux_out) {
   JXL_ASSERT(x0 + xsize <= frame_data.xsize);
   JXL_ASSERT(y0 + ysize <= frame_data.ysize);
+  const FrameHeader& frame_header = mutable_frame_header;
+  PassesSharedState& shared = enc_state.shared;
+  shared.metadata = metadata;
+  if (enc_state.streaming_mode) {
+    shared.frame_dim.Set(xsize, ysize, /*group_size_shift=*/1,
```

Comment:
I wonder if we should rename `shared.frame_dim` (later), as it seems to indicate the size of the currently-processed patch now :)

Reply:
This is also used in non-streaming mode and in decoder, so choosing a name that describes all uses might be tricky (e.g. frame_or_patch_dim seems overly verbose)

Reply:
I'd say `patch_dim` is fine (and in non-streaming mode, `patch` = `frame`)

Reply:
ok, but as you said it would be better in another PR